odoo/addons/base/models/ir_profile.py

```diff
@@ -108,3 +118,25 @@ def set_profiling(self, profile=None, collectors=None, params=None):
             'collectors': request.session.profile_collectors,
             'params': request.session.profile_params,
         }
+
+
+class EnableProfilingWizard(models.TransientModel):
+    _name = 'base.enable.profiling.wizard'
+    _description = "Enable profiling for some time"
+
+    duration = fields.Selection([
+        ('minutes_5', "5 Minutes"),
+        ('hours_1', "1 Hour"),
+        ('days_1', "1 Day"),
+        ('months_1', "1 Month")
+    ], string="Enable profiling for")
+    expiration = fields.Datetime("Enable profiling until", compute='_compute_expiration', store=True, readonly=False)
+
+    @api.depends('duration')
+    def _compute_expiration(self):
+        for record in self:
+            unit, quantity = (record.duration or 'days_0').split('_')
+            record.expiration = fields.Datetime.now() + relativedelta(**{unit: int(quantity)})
+
+    def submit(self):
+        self.env['ir.config_parameter'].set_param('base.profiling_enabled_until', self.expiration)
```

Comment:
Does this close the wizard form?

Reply:
Yes

Reply:
Still seems like a good idea to return either `False` or a really real `ir.actions.close` action, to avoid blowing up an eventual XML-RPC call.